net/raft/raft.go

```diff
@@ -855,8 +855,8 @@ func (sv *Service) getSnapshot() *raftpb.Snapshot {
 		panic(err)
 	}
 	sv.confMu.Lock()
-	defer sv.confMu.Unlock()
 	snap, err := sv.raftStorage.CreateSnapshot(index, &sv.confState, data)
+	sv.confMu.Unlock()
```

Comment:
In applyEntry we lock applyMu before locking confMu, so we need to be careful to always lock them in that order to prevent deadlocks.

It's probably very unlikely, but if in the future we needed to acquire applyMu in this function we might forget that confMu is still locked until the function returns.